crypto/rsa/rsa_pmeth.c

```diff
@@ -153,7 +154,9 @@ static int pkey_rsa_sign(EVP_PKEY_CTX *ctx, unsigned char *sig,
             if (ret <= 0)
                 return ret;
             ret = sltmp;
-        } else if (rctx->pad_mode == RSA_X931_PADDING) {
+        } else
+#endif /* FIPS_MODE */
+        if (rctx->pad_mode == RSA_X931_PADDING) {
```

Comment:
Why are you changing this file at all?  It should *never* become part of the FIPS module, this is a legacy backend.

If there are things here that are missing in `providers/implementations/signature/rsa.c`, then they should be added there.